devicetypes/smartthings/fibaro-heat-controller-with-sensor.src/fibaro-heat-controller-with-sensor.groovy

```diff
@@ -0,0 +1,304 @@
+/**
+ *	Copyright 2018 SmartThings
+ *
+ *	Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *	in compliance with the License. You may obtain a copy of the License at:
+ *
+ *		http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *	Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+
+ *	on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *	for the specific language governing permissions and limitations under the License.
+ *
+ */
+metadata {
+	definition (name: "Fibaro Heat Controller With Sensor", namespace: "smartthings", author: "Samsung", mnmn: "SmartThings", vid: "SmartThings-smartthings-Z-Wave_Battery_Thermostat", ocfDeviceType: "oic.d.thermostat") {
+		capability "Thermostat Mode"
+		capability "Refresh"
+		capability "Battery"
+		capability "Thermostat Heating Setpoint"
+		capability "Health Check"
+		capability "Thermostat"
+
+		command "setThermostatSetpointUp"
+		command "setThermostatSetpointDown"
+		command "switchMode"
+	}
+
+	tiles(scale: 2) {
+		multiAttributeTile(name:"thermostat", type:"general", width:6, height:4, canChangeIcon: false)  {
+			tileAttribute("device.heatingSetpoint", key: "VALUE_CONTROL") {
+				attributeState("VALUE_UP", action: "setThermostatSetpointUp")
+				attributeState("VALUE_DOWN", action: "setThermostatSetpointDown")
+			}
+			tileAttribute("device.thermostatMode", key: "PRIMARY_CONTROL") {
+				attributeState("off", action:"switchMode", nextState:"...", icon: "st.thermostat.heating-cooling-off", label: '${currentValue}')
+				attributeState("auto", action:"switchMode", nextState:"...", icon: "st.thermostat.auto", label: '${currentValue}')
+				attributeState("heat", action:"switchMode", nextState:"...", icon: "st.thermostat.heat", label: '${currentValue}')
+			}
+		}
+
+		valueTile("battery", "device.battery", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "battery", label: 'Battery:\n${currentValue}%', unit: "%"
+		}
+
+		standardTile("refresh", "command.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "refresh", label: 'refresh', action: "refresh.refresh", icon: "st.secondary.refresh-icon"
+		}
+		main "thermostat"
+		details(["thermostat", "battery", "refresh"])
+	}
+}
+
+def installed() {
+	initialize()
+}
+
+def updated() {
+	initialize()
+}
+
+def initialize() {
+	if(!childDevices) {
+		addChild()
+	}
+	def supportedModes = ["off", "heat", "auto"]
+	state.supportedModes = supportedModes
+	sendEvent(name: "supportedThermostatModes", value: supportedModes, displayed: false)
+
+	sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+	sendHubCommand(setThermostatMode("off"))
```

Comment:
why?

Reply:
When paired device tends to have "auto" mode active, even if temperature sensor isn't operating, so I decided to send this command to prevent running a mode which isn't supported.